src/BenchmarkDotNet/Engines/Engine.cs

```diff
@@ -135,9 +143,15 @@ public Measurement RunIteration(IterationData data)
             bool isOverhead = data.IterationMode == IterationMode.Overhead;
             var action = isOverhead ? OverheadAction : WorkloadAction;
 
-            if(!isOverhead)
+            if (!isOverhead)
                 IterationSetupAction();
 
+            if (data.IterationMode == IterationMode.Workload && data.IterationStage == IterationStage.Actual)
```

Comment:
It's better to use `IsWorkload()` extension method here (and introduce a similar method for the Actual Stage).

Reply:
I've used this method, but please see the comment below.